src/dispatcher.rs

```diff
@@ -158,12 +174,22 @@ impl TxnDispatcher for Dispatcher {
         let outputs = TxnBatch::new(
             calls
                 .iter()
-                .map(|call| match call {
-                    Ok(call) => self
-                        .encode_response(call, &mut ctx)
-                        .or_else(|err| -> Fallible<Vec<u8>> { Ok(self.serialize_error(&err)) })
-                        .unwrap(), // Can't fail because the error is always mapped into an Ok() above.
-                    Err(err) => self.serialize_error(err),
+                .map(|call| {
+                    if self
+                        .abort_batch
+                        .as_ref()
+                        .map(|b| b.load(Ordering::SeqCst))
+                        .unwrap_or(false)
+                    {
+                        return self.serialize_error(&anyhow!("batch aborted"));
+                    }
+                    match call {
+                        Ok(call) => self
+                            .encode_response(call, &mut ctx)
+                            .or_else(|err| -> Result<Vec<u8>> { Ok(self.serialize_error(&err)) })
+                            .unwrap(), // Can't fail because the error is always mapped into an Ok() above.
```

Comment:
that's what [unwrap_or_else](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else) does :)